src/Avalonia.Input/PenDevice.cs

```diff
@@ -0,0 +1,415 @@
+        private PixelPoint? _position; 
+
+        public PenDevice(Pointer? pointer = null)
+        {
+            _pointer = pointer ?? new Pointer(Pointer.GetNextFreeId(), PointerType.Pen, true);
+        }
+        
+        /// <summary>
+        /// Gets the control that is currently capturing by the mouse, if any.
+        /// </summary>
+        /// <remarks>
+        /// When an element captures the mouse, it receives mouse input whether the cursor is 
+        /// within the control's bounds or not. To set the mouse capture, call the 
+        /// <see cref="Capture"/> method.
+        /// </remarks>
+        [Obsolete("Use IPointer instead")]
+        public IInputElement? Captured => _pointer.Captured;
+
+        public bool IsEraser { get; set; }
+        public bool IsInverted { get; set; }
+        public bool IsBarrel { get; set; }
+        public int XTilt { get; set; }
+        public int YTilt { get; set; }
+        public uint Pressure { get; set; }
+        public uint Twist { get; set; }
+
+        /// <summary>
+        /// Captures mouse input to the specified control.
+        /// </summary>
+        /// <param name="control">The control.</param>
+        /// <remarks>
+        /// When an element captures the mouse, it receives mouse input whether the cursor is 
+        /// within the control's bounds or not. The current mouse capture control is exposed
+        /// by the <see cref="Captured"/> property.
+        /// </remarks>
+        public void Capture(IInputElement? control)
+        {
+            _pointer.Capture(control);
+        }
+
+        /// <summary>
+        /// Gets the mouse position relative to a control.
+        /// </summary>
+        /// <param name="relativeTo">The control.</param>
+        /// <returns>The mouse position in the control's coordinates.</returns>
+        public Point GetPosition(IVisual relativeTo)
+        {
+            relativeTo = relativeTo ?? throw new ArgumentNullException(nameof(relativeTo));
+
+            if (relativeTo.VisualRoot == null)
+            {
+                throw new InvalidOperationException("Control is not attached to visual tree.");
+            }
+
+#pragma warning disable CS0618 // Type or member is obsolete
+            var rootPoint = relativeTo.VisualRoot.PointToClient(_position ?? new PixelPoint(-1, -1));
+#pragma warning restore CS0618 // Type or member is obsolete
+            var transform = relativeTo.VisualRoot.TransformToVisual(relativeTo);
+            return rootPoint * transform!.Value;
+        }
+
+        public void ProcessRawEvent(RawInputEventArgs e)
+        {
+            if (!e.Handled && e is RawPointerEventArgs margs)
+                ProcessRawEvent(margs);
+        }
+
+        public void TopLevelClosed(IInputRoot root)
+        {
+            ClearPointerOver(this, 0, root, PointerPointProperties.None, KeyModifiers.None);
+        }
+
+        public void SceneInvalidated(IInputRoot root, Rect rect)
```

Comment:
I heavily suspect that we'll get problems with combining mouse and pen input if we handle pointerover separately like that. 
I'd suggest to keep pointerover out of that PR and refactor it into some more unified system later.

Reply:
I need a pointer over to work with pen. Conflicts of mouse and pen - is a super corner case. User uses pen or mouse in most cases.

Reply:
>is a super corner case


Some wacom tablets come with a touchpad. Switching between pen and touch _will_ be frequent.